libraries/aws_backend.rb

```diff
@@ -230,7 +230,21 @@ def validate_parameters(allow: [], required: nil, require_any_of: nil)
   end
 
   def failed_resource?
-    @failed_resource
+    @failed_resource ||= false
+  end
+
+  def tags
+    return if failed_resource?
+    # Prevent `NoMethod` error if AWS API does not return `tags` property when it is not applied to the resource.
+    # E.g., subnets.
+    {}
+  end
+
+  def name
+    return unless tags
+    return tags['Name'] if tags.is_a?(Hash)
+    # tags might be in the original format: [{:key=>"Name", :value=>"aws-linux-ubuntu-vm"}], e.g in EC2
+    tags.select { |tag| tag[:key] == 'Name' }.first&.dig(:value)
   end
```

Comment:
Not sure how I feel about this being moved to aws_backend. What about creating a helper method class or something along those lines to prevent this from potentially becoming cluttered.

Reply:
Happy to do that.
The reason I put it in the superclass is that all the existing resources can have these new methods defined without changing the subclasses.

Reply:
@skpaterson, what is your opinion on that?

Reply:
hey @omerdemirok - I'm good with proceeding as is for now and we can look at refactoring as @rmoles suggests afterwards.